### PR TITLE
Update `AttestationClient::AttestTpm` API to match existing `AttestOpenEnclave` and `AttestSgxmEnclave` 

### PR DESCRIPTION
**Pull request by penglimsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/attestation/azure-security-attestation/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 ### Breaking Changes
 
+- Changed `AttestationClient::AttestTpm` to match `AttestOpenEnclave` and `AttestSgxmEnclave`
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/CHANGELOG.md:9]: `AttestSgxEnclave`
```suggestion
- Changed `AttestationClient::AttestTpm` to match `AttestOpenEnclave` and `AttestSgxEnclave` 
```

File: sdk/attestation/azure-security-attestation/src/attestation_client.cpp [hunk 2]
```diff
@@ -212,7 +213,7 @@ Azure::Response<TpmAttestationResult> AttestationClient::AttestTpm(
     // Send the request to the service.
     auto response
         = AttestationCommonRequest::SendRequest(*m_pipeline, request, tracingContext.Context);
-    std::string returnedBody(TpmDataSerializer::Deserialize(response));
+    auto returnedBody = TpmDataSerializer::Deserialize(response);
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/src/attestation_client.cpp:216]: C++ guidelines discourage the use of assignment initialization in favor of either ctor syntax (`returnedBody(TpmDataSerializer::Deserialize(response))`) or initializer syntax (`returnedBody{TpmDataSerializer::Deserialize(response)}`).

**Review by LarryOsterman — CHANGES_REQUESTED** [review]

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp [hunk 3]
```diff
@@ -225,24 +225,21 @@ namespace Azure { namespace Security { namespace Attestation {
     /**
      * @brief Perform a single leg
      *
-     * Processes attestation evidence from a VBS enclave, producing an attestation result.
+     * Attest a TPM based enclave
      *
      * The TPM attestation protocol is defined
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp:230]: QQ: The documentation in the "virtualization-based-security-protocol" is incorrect (it leaves off the "payload" JSON object). Should that mistake be reflected in the documentation for the `AttestTPM` API, or should it be fixed on the protocol documentation?

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp [hunk 4]
```diff
@@ -225,24 +225,21 @@ namespace Azure { namespace Security { namespace Attestation {
     /**
      * @brief Perform a single leg
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp:226]: This brief description (which is the highlighted opening paragraph) is not a complete sentence.

**Review by LarryOsterman — APPROVED** [review]

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp (hunk 3 above)
Comment (penglimsft) [reply sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp:230]: We will address the issue in the documentation. Will create a task to track it

**Comment by gkostal** [comment]
LGTM.  How is the version number updated to indicate this is a breaking change (i.e. 2.* I believe)?

**Review by gkostal — APPROVED** [review]

**Comment by LarryOsterman** [comment]
> LGTM. How is the version number updated to indicate this is a breaking change (i.e. 2.* I believe)?

When the attestation team wants to release a new version of the attestation service, they run the prepare-release script. In that script, the attestation team specifies that they want the new version to be 2.0.

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**